core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -73,6 +81,8 @@ class KafkaHealthcheck(private val brokerId: Int,
     @throws(classOf[Exception])
     def handleStateChanged(state: KeeperState) {
       // do nothing, since zkclient will do reconnect for us.
+      if(state == KeeperState.Expired)
+        shutdown()
```

Comment:
If the session expires, then the ephemeral will be gone and I'm trying to prevent is that we recreate the ephemeral across sessions. I don't think we need to do anything else upon other events, since we only care in this case about the session expiration, but maybe I'm missing something. Is there any specific concern you have, like a scenario you think this could be missing?